### PR TITLE
Sealing update

### PR DESCRIPTION
**Pull request by casssi**
## Description

Sealing added

### CODE REVIEW

File: server/blindai_sgx/config.toml [hunk 1]
```diff
@@ -14,4 +14,7 @@ client_to_enclave_attested_url = "https://0.0.0.0:50051"
 max_model_size = 524288000
 
 # Max input size in bytes (default value: 500mb)
-max_input_size = 524288000
\ No newline at end of file
+max_input_size = 524288000
+
+# Set the path
+models_path = "/workspaces/blindai/server/bin/models"
```
Comment (cchudant) [inline server/blindai_sgx/config.toml:20]: set the default value as "./models"

File: client/blindai/client.py [hunk 2]
```diff
@@ -570,12 +572,16 @@ def upload_model(
...
         """
-
+        #print(model_name)
         response = None
 
         if model_name is None:
             model_name = os.path.basename(model)
 
+
+        print(model_name)
```
Comment (cchudant) [inline client/blindai/client.py:582]: remove prints

File: client/blindai/client.py [hunk 3]
```diff
@@ -560,6 +561,7 @@ def upload_model(
             dtype_out (ModelDatumType, optional): The type of the model output data (f32 by default). Defaults to ModelDatumType.F32.
             sign (bool, optional): Get signed responses from the server or not. Defaults to False.
             model_name (Optional[str], optional): Name of the model.
+            sealed (bool): If the model have to be sealed or not
```
Comment (cchudant) [inline client/blindai/client.py:564]: update docstring

File: server/blindai_sgx/src/client_communication.rs [hunk 4]
```diff
@@ -117,6 +118,7 @@ impl Exchange for Exchanger {
                 }
 
                 sign = model_proto.sign;
+                save_model= model_proto.save_model;
```
Comment (cchudant) [inline server/blindai_sgx/src/client_communication.rs:121]: you will need to run rust-fmt to format the files

File: server/blindai_sgx/src/lib.rs [hunk 5]
```diff
@@ -139,13 +140,40 @@ async fn main(
...
-        ModelStore::new().into(),
+        model_store.clone(),
         my_identity.clone(),
         network_config.max_model_size,
         network_config.max_input_size,
     );
 
+    ModelStore::startup_unseal(model_store.clone());
+
+    /*let mut models_path = std::env::current_dir()?;
```
Comment (cchudant) [inline server/blindai_sgx/src/lib.rs:154]: remove this big comment now i guess?

File: tests/test_covidnet.py [hunk 6]
```diff
@@ -103,6 +103,51 @@ def test_multiple(self):
...
+                model=model,
+                shape=(1, 480, 480, 3),
+                dtype=ModelDatumType.F32,
+                sign=True,
+                save_model=True,
+                model_name="Salut"
+            )
+
+            if not self.simulation and os.getenv("BLINDAI_DUMPRES") is not None:
+                upload_response.save_to_file("./client/tests/exec_upload.proof")
```
Comment (cchudant) [inline tests/test_covidnet.py:130]: remove the save_to_file things in this test, we only need it in 1 test

File: tests/test_covidnet.py [hunk 7]
```diff
@@ -103,6 +103,51 @@ def test_multiple(self):
...
+
+            response = client.run_model(
+                upload_response.model_id,
+                flattened_img,
+                sign=True,
+            )
+
+        if not self.simulation and os.getenv("BLINDAI_DUMPRES") is not None:
```
Comment (cchudant) [inline tests/test_covidnet.py:138]: same

File: server/blindai_sgx/src/model_store.rs [hunk 8]
```diff
@@ -36,17 +52,57 @@ impl ModelStore {
 
     pub fn add_model(
         &self,
+        //model_path: &Path,
         model_bytes: &[u8],
         input_facts: Vec<Vec<usize>>,
         model_name: Option<String>,
+        //model_id: Uuid,
```
Comment (cchudant) [inline server/blindai_sgx/src/model_store.rs:59]: remove comments like this, keep the code as clean as possible :)

**Review by cchudant — COMMENTED** [review]
this is good, my issues are mostly regarding code style/formatting, and we will see whether the CI tests fail

**Comment by JoFrost** [comment]
Closing as part of the input-output-rework branch.

**Pull request closed without merging**